lib/widgets/compose_box.dart

```diff
@@ -1032,25 +1048,190 @@ class _ComposeBoxLayout extends StatelessWidget {
+            ),
+          ),
           const SizedBox(width: 8),
-          sendButton,
+          widget.sendButton,
         ]),
         Theme(
           data: themeData.copyWith(
             iconTheme: themeData.iconTheme.copyWith(color: colorScheme.onSurfaceVariant)),
           child: Row(children: [
-            _AttachFileButton(contentController: contentController, contentFocusNode: contentFocusNode),
-            _AttachMediaButton(contentController: contentController, contentFocusNode: contentFocusNode),
-            _AttachFromCameraButton(contentController: contentController, contentFocusNode: contentFocusNode),
-          ])),
-      ]));
+            IconButton(
+              icon: Icon(isPreviewMode ? Icons.visibility_off : Icons.visibility),
+              tooltip: isPreviewMode
+                  ? 'Switch to Edit Mode'
+                  : 'Preview Content',
+              onPressed: togglePreview,
+            ),
+            _AttachFileButton(
+              contentController: widget.contentController,
+              contentFocusNode: widget.contentFocusNode,
+            ),
+            _AttachMediaButton(
+              contentController: widget.contentController,
+              contentFocusNode: widget.contentFocusNode,
+            ),
+            _AttachFromCameraButton(
+              contentController: widget.contentController,
+              contentFocusNode: widget.contentFocusNode,
+            ),
+
+          ]),
+        ),
+      ]),
+    );
   }
 }
 
+
+
+class PreviewBox extends StatelessWidget {
+  const PreviewBox({super.key, required this.content});
+
+  final String content;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+
+    return Container(
+      constraints: const BoxConstraints(
+        minHeight: _sendButtonSize - 2 * _inputVerticalPadding,
+        maxHeight: 200,
+      ),
+      width: double.infinity,
+      padding: const EdgeInsets.symmetric(
+        horizontal: 10.0,
+        vertical: _inputVerticalPadding + 4,
+      ),
+      decoration: BoxDecoration(
+        color: colorScheme.surface,
+        borderRadius: BorderRadius.circular(4.0),
+      ),
+      child: SingleChildScrollView(
+        child: content.isEmpty
+            ? Text(
+                '(No content to preview)',
+                style: weightVariableTextStyle(context).merge(
+                  const TextStyle(
+                    fontSize: 16,
+                    height: 22 / 16,
+                  ),
+                ),
+              )
+            : RichText(
+                text: _buildStyledText(content, context),
+              ),
+      ),
+    );
+  }
+
+  InlineSpan _buildStyledText(String text, BuildContext context) {
+    final boldPattern = RegExp(r'\*\*(.*?)\*\*');
+    final italicPattern = RegExp(r'\*(.*?)\*');
+    final codePattern = RegExp(r'```(.*?)```', dotAll: true); // Match ```code```
```

Comment:
Hmm. I don't think parsing the message content with regular expressions is the right approach. This will eventually leads to the complexity of re-implementing Zulip's markdown processor from scratch, if we want to have full support of all message content features. A correct implementation should use the [render-message](https://zulip.com/api/render-message) endpoint, and reuse Zulip's Content parser.